### DISCUSSION TITLE
Difference between tree-sitter-cli and neovim highlighting priority

### DISCUSSION THREAD

**Discussion by luckasRanarison** [nvim-treesitter/nvim-treesitter#5031] [Q&A] [answered]
Hey there! I've created a parser and highlight queries for my toy programming language and I got an issue, the highlight priority in the cli and neovim are different, tree-sitter-cli captures the first highlight and neovim the last one. On the top left the actual neovim highlighting and on the bottom left the cli highlighting:

![Screenshot from 2023-06-28 12-32-26](https://github.com/nvim-treesitter/nvim-treesitter/assets/101930730/c070cc6f-9e6c-4537-a448-3189cd5e29a2)

Inspection:

![Screenshot from 2023-06-28 12-38-47](https://github.com/nvim-treesitter/nvim-treesitter/assets/101930730/e2330e31-2523-4ced-8858-f5058d2bb2ba)

**Comment by clason** [comment, marked as answer]
Yes, that is intentional and due to the different precedence of query order:
1. tree-sitter-cli has "first match wins" (stop processing queries at first match);
2. Neovim has -- sort of -- "last match wins" (process all queries and apply them in order; last one wins by "painting over" previous ones).

**Reply by luckasRanarison** [reply]
Thanks for the quick reply!

**Reply by clason** [reply]
Would be great though if `tree-sitter` CLI had a flag that could toggle the behavior (hint, hint ;))
